File: tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/FloatArrayMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class FloatArrayMapHelper {
        public static int getByteArrayLength(Dictionary<String, float[]> floatArrayMap) {
            int blength = 0;
            foreach (KeyValuePair<String, float[]> item in floatArrayMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                float[] values = item.Value;
                blength += TransferUtil.getLengthOfFloatArray(values);
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, float[]> floatArrayMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, float[]> item in floatArrayMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_FLOATARRAY);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                float[] values = item.Value;
                touts.writeFloatArray(values);
            }
        }
    }
}

File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/FloatArrayMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class FloatArrayMapHelper {
	
	public static int getByteArrayLength(Map<String, float[]> floatArrayMap){
		int blength = 0;
		Iterator<String> keyIt = floatArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			float[] values = floatArrayMap.get(key);
			blength += TransferUtil.getLengthOfFloatArray(values);
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, float[]> floatArrayMap, TransferOutputStream touts) {
		Iterator<String> keyIt = floatArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_FLOATARRAY);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			float[] values = floatArrayMap.get(key);
			touts.writeFloatArray(values);
		}
	}
}
